### ISSUE TITLE
Invalid example for custom compression codecs

### ISSUE THREAD

**Issue by Nevon** [tulios/kafkajs#1100]
**Describe the bug**
[In the documentation for custom compression codecs](https://kafka.js.org/docs/producing#a-name-compression-other-a-other), there's an example where a custom compression codec is registered like so:

```ts
const MyCustomSnappyCodec = {
    async compress(encoder) {
        return someCompressFunction(encoder.buffer)
    },

    async decompress(buffer) {
        return someDecompressFunction(buffer)
    }
}

const { CompressionTypes, CompressionCodecs } = require('kafkajs')
CompressionCodecs[CompressionTypes.Snappy] = MyCustomSnappyCodec
```

But in fact, we expect the value to be a function returning the codec, allowing for lazy initialization. The Typescript types are correct (although they say the function should return `any`, which is not ideal)  https://github.com/tulios/kafkajs/blob/06f8d2287f14f0ca9032122260b8bd3a34ba1eb0/types/index.d.ts#L915-L920

**Expected behavior**
The documentation should show that the expected value is a function returning `MyCustomSnappyCodec`.

**Comment by prithvijit-dasgupta** [comment]
Hey, I want to work this @Nevon. Basically, the documentation needs another line with a function returning the codec right and the function plugged in to `CompressionCodecs`?

**Comment by Nevon** [comment]
Yup. It's a very small change. You could even change it to just use an arrow function inline instead of declaring a function.

**Comment by Nevon** [comment]
Fixed by #1102
